OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -28,15 +28,30 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
     private var messageQueue : NSMutableOrderedSet
     private var pendingTickleIds : NSMutableOrderedSet
 
+    private var displayMode : InAppDisplayMode
     private var currentMessage : InAppMessage?
 
-    override init() {
+    init(displayMode: InAppDisplayMode) {
         self.messageQueue = NSMutableOrderedSet.init(capacity: 5)
         self.pendingTickleIds = NSMutableOrderedSet.init(capacity: 2)
         self.currentMessage = nil
+        
+        self.displayMode = displayMode
 
         super.init()
     }
+    
+    func setDisplayMode(_ mode: InAppDisplayMode) {
+        runOnMainThreadSync {
```

Comment:
Using this to serialise writes to the value.

Reads 'usually' appear from main thread, but not traced if they _always_ do yet.

Reply:
Reads are now done through the accessor on the main thread.